File: system-test/typescript/src/testkit/dsl/port/myShop/then/steps/then-coupon.ts
export interface ThenCoupon extends PromiseLike<void> {
  and(): this;
  hasDiscountRate(discountRate: number): this;
  isValidFrom(validFrom: string): this;
  isValidTo(validTo: string): this;
  hasUsageLimit(usageLimit: number): this;
  hasUsedCount(expectedUsedCount: number): this;
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/then/steps/ThenCoupon.java
package com.mycompany.myshop.testkit.dsl.port.then.steps;

import com.mycompany.myshop.testkit.dsl.port.then.steps.base.ThenStep;

public interface ThenCoupon extends ThenStep<ThenCoupon> {
    ThenCoupon hasDiscountRate(double discountRate);

    ThenCoupon isValidFrom(String validFrom);

    ThenCoupon isValidTo(String validTo);

    ThenCoupon hasUsageLimit(int usageLimit);

    ThenCoupon hasUsedCount(int expectedUsedCount);
}
